spec/console/console_spec.rb

```diff
@@ -32,24 +32,40 @@
     EXPECTED_OUTPUT = "| X |   | O | "
     @game.board.check_tile(1, 0, @player1)
     @game.board.check_tile(1, 2, @player2)
-    @console.print_row(@game.board, 1)
+    @console.send(:print_row, @game.board, 1)
 
     expect(@output.string).to eql(EXPECTED_OUTPUT)
   end
 
-  it "Gets the game mode option from User input" do
+  it "Gets the game mode option from user input" do
     input = StringIO.new("1")
     @console = Console.new(@output, input)
 
-    expect(@console.get_menu_option).to eql("1")
+    expect(@console.send(:get_menu_option)).to eql("1")
   end
 
   it "When a user inputs an incorrect game option, it gets reprompted" do
     input = StringIO.new("g\n1")
     @console = Console.new(@output, input)
 
-    expect(@console.get_menu_option).to eql("1")
+    expect(@console.send(:get_menu_option)).to eql("1")
   end
 
-  #TODO Test play method - The loop. Should we extract that logic? Yes.
+  it "Initializes the game with user input" do
+    input = StringIO.new("1")
+    @console = Console.new(@output, input)
+    @console.send(:init_game)
+
+    expect(@console.game).not_to eql(nil)
+  end
+
+  it "Makes moves until one player has won or it's a draw" do
+    output = StringIO.new
+    input = StringIO.new(GameOptions::MACHINE_VS_MACHINE)
+    @console = Console.new(output, input)
```

Comment:
Why is console an instance variable (ivar)?

Reply:
Because in some cases I set it up in the `before` hook. Is that a bad practice? I could be consistent and always set it up in each individual test, but then I would have a lot of lines duplicated and shared between methods.

Reply:
You can use let in rspec and then refer to it through the getter, i.e. console.whatever instead of @console.whatever. I prefer not to refer to ivars in my ruby code as much as possible to allow for easier refactoring. If you change where that thing comes from, you can just create the method, i.e., def console; end; and none of the code that uses it needs to change. This is nitpick territory, I just dislike it when code doesn't look as consistent as possible.